jsk_recognition_utils/src/geo/line.cpp

```diff
@@ -198,8 +198,60 @@ namespace jsk_recognition_utils
              origin_[0], origin_[1], origin_[2]);
   }
 
-  void Line::point(double alpha, Eigen::Vector3f& output)
+  void Line::point(double alpha, Eigen::Vector3f& output) const
   {
     output = alpha * direction_ + origin_;
   }
+
+  void Line::commonPerpendicularPoints(const Line& other, Eigen::Vector3f& output1, Eigen::Vector3f& output2) const
+  {
+    if(isParallel(other))
+      return;
+    Eigen::Vector3f vec12 = (other.origin_ - origin_);
+    double d12 = direction_.dot(other.direction_);
+    double alpha1 = (vec12.dot(direction_) - (vec12.dot(other.direction_) * d12)) / (1.0 - d12 * d12);
+    double alpha2 = (- vec12.dot(other.direction_) + (vec12.dot(direction_) * d12)) / (1.0 - d12 * d12);
+    point(alpha1, output1);
+    other.point(alpha2, output2);
+  }
+
+  Line::Ptr Line::commonPerpendicular(const Line& other) const
```

Comment:
Thank you for advice!
I added nearestPoint function instead of commonPerpendicularPoints and reuse it in the other functions